restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCConnection.java

```diff
@@ -560,10 +563,19 @@ void initializeWebrtc(boolean videoEnabled)
         for (String permission : MANDATORY_PERMISSIONS) {
             if (context.checkCallingOrSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                 logAndToast("Permission " + permission + " is not granted");
-                // TODO: return error to RCConnection listener
+                listener.onPermissionNotGranted(this, permission);
```

Comment:
Oh, but of course! I thought you were referring to the RCConnection state, my bad. Yes please update the RCDevice state to be READY again right before you notify the App that it has no suitable permissions, so that subsequent requests can be made without issues.